moma_controllers/moma_ocs2/include/moma_ocs2/MobileManipulatorPinocchioMapping.h

```diff
@@ -39,19 +39,20 @@ template <typename SCALAR>
 class MobileManipulatorPinocchioMapping final : public PinocchioStateInputMapping<SCALAR> {
  private:
   BaseType baseType_;
+  size_t armInputDim_;
  public:
   using Base = PinocchioStateInputMapping<SCALAR>;
   using typename Base::matrix_t;
   using typename Base::vector_t;
 
-  MobileManipulatorPinocchioMapping(BaseType baseType = BaseType::none) : baseType_(baseType) {};
+  MobileManipulatorPinocchioMapping(const size_t armInputDim, const BaseType baseType = BaseType::none) : armInputDim_(armInputDim), baseType_(baseType) {};
   ~MobileManipulatorPinocchioMapping() override = default;
   MobileManipulatorPinocchioMapping<SCALAR>* clone() const override { return new MobileManipulatorPinocchioMapping<SCALAR>(*this); }
 
   vector_t getPinocchioJointPosition(const vector_t& state) const override { return state; }
 
   vector_t getPinocchioJointVelocity(const vector_t& state, const vector_t& input) const override {
-    vector_t dxdt(STATE_DIM);
+    vector_t dxdt(STATE_DIM(armInputDim_));
```

Comment:
Why is this initialized with dim `armInputDim_` ? The dimension does not account for the base